[Feat] Add @OptionFunction decorator

gund opened the pull request:
Implementation of RFC #28 (closes #28)

- [x] Decode/validate serialized function
- [x] Decode/validate function metadata (`{args: string[], body: string}`)
- [x] Resolve arguments via local `Injector`
- [x] Capture errors from functions and forward to `ErrorStrategy`
- [x] Handle arguments which starts with `$`

PxyUp commented on libs/core/src/lib/util.ts:
```diff
@@ -0,0 +1,54 @@
+    arr.push(...group);
+  }
+
+  return !arr.length ? null : arr;
```
May be here return always arr, instead return null in case arr.length === 0

gund replied:
This is done to copy same behavior of native `match/exec` methods, if nothing was found - `null` will be returned.

gund commented:
Implementation is complete. More unit tests required to verify that it's working.

gund commented:
Implementation is complete, need more unit tests.

gund commented:
Implementation and unit tests are complete now.

/cc @RyuuGan @baruchvlz please review before merging.

baruchvlz commented:
Shouldn't we move the UI stuff to another PR? I get that it doesn't really matter much, but you know, keep things separated.

gund commented:
@baruchvlz yeah you are right. We should, but here I just wanted to quickly verify in demo app that it is working end-to-end so I had to add just that little bit.

I won't do it again =)

gund commented:
I've added a new prop to dynamic component's that allows to set any [event handlers on host components](https://github.com/orchestratora/orchestrator/pull/29/files#diff-e781977c32f0e7ab20244dfc0e965aabR18).

Thanks to this I removed `click` from ui-web-button component config.
